server/server.go

```diff
@@ -276,7 +285,10 @@ func New(cfg Config) (*Server, error) {
 			RouteMatcher: router,
 		},
 	}
-
+	httpMiddleware := []middleware.Interface{}
+	if !cfg.DoNotAddDefaultHTTPMiddleware {
+		httpMiddleware = append(httpMiddleware, defaultHTTPMiddleware...)
+	}
 	httpMiddleware = append(httpMiddleware, cfg.HTTPMiddleware...)
```

Comment:
or you could put the `DoNotAddDefaultHTTPMiddleware` condition on this line, so you either do an append or take `cfg.HTTPMiddleware` unaltered

Reply:
That indeed reads better.